packages/msgraph/src/Utils.js

```diff
@@ -135,3 +131,67 @@ export const request = async (urlString, params = {}, method = 'GET') => {
 
   return data;
 };
+
+function makeAuthHeader(accessToken) {
+  return accessToken ? `Bearer ${accessToken}` : null;
+}
+
+const defaultSheetOptions = {
+  bookType: 'xlsx',
+  wsName: 'Sheet',
+};
+
+/**
+ * The function `sheetToBuffer` takes in rows, options and optional callback, It creates a workbook
+ * and worksheet using the rows, appends the worksheet to the workbook, and returns the workbook as a
+ * buffer.
+ * @public
+ * @example
+ * <caption>Create a buffer containing excel file with `xlsx` output format  </caption>
+ * sheetToBuffer('$.data[*]', {
+ *  wsName: 'Invalid Grant Codes',
+ *  bookType: 'xlsx',
+ * });
+ * @param rows - The `rows` parameter is an array of objects representing the data to be written to the
```

Comment:
The docs on this parameter aren't very clear. Also it looks lie rows takes a string with a path to "an array objects...", rather than an array of objects? Can we pass data directly?

I'm happy to wave this through as its just in this adaptor. It's the kind of thing that would get a lot more scrutiny if we were to merge it into common.

Reply:
Also one of the strange things about this function is that it doesn't really take a sheet. It looks like it just takes an array of json objects, converts them into a sheet (actually in your job aren't you converting into an array?) then turning THAT into a buffer ready for upload.

Again, because it's not going into common I'm happy to not look too closely at this

Reply:
Your right the function takes an Array of Objects Data then convert it to buffer ready for upload, What shouldbe best name for this function @josephjclark

Reply:
Well what it does it `jsonToSheet`, it's just that the sheet is then represented as a buffer.

Let's leave the name for now - this function is really designed just for this one job and I don't know how well it'll scale beyond that.

We do need to fix the docs though - `[daoptionsta.bookType]`

Reply:
Oh you did that already I see :+1: